src/S3FileCache.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SmartFrame\IP2Location;
+
+use Aws\S3\S3ClientInterface;
+use GuzzleHttp\Psr7\Stream;
+use Psr\Http\Message\StreamInterface;
+use SmartFrame\IP2Location\Contracts\FileCacheInterface;
+use SmartFrame\IP2Location\Exception\EmptyFileException;
+
+class S3FileCache implements FileCacheInterface
+{
+    private S3ClientInterface $s3Client;
+    private string $bucket;
+    private string $prefix;
+
+    public function __construct(S3ClientInterface $s3Client, string $bucket, string $prefix)
+    {
+        $this->s3Client = $s3Client;
+        $this->bucket = $bucket;
+        $this->prefix = $prefix;
+    }
+
+    public function read(string $path): StreamInterface
+    {
+        $cmd = $this->s3Client->getCommand('GetObject', [
+            'Bucket' => $this->bucket,
+            'Key' => $this->prefix . $path,
+        ]);
+        $request = $this->s3Client->createPresignedRequest($cmd, '+1 minute');
+
+        return new Stream(fopen((string) $request->getUri(), 'rb'));
+    }
+
+    public function write(
+        string $path,
+        Stream $stream,
+        string $mimeType = 'application/octet-stream',
+        $class = 'STANDARD'
```

Comment:
please use typing and a proper variable name 
```suggestion
        string $storageClass = 'STANDARD'
```
Consider using STANDARD_IA since file size can be above 131072B